R/Surv_CDISC.R

```diff
@@ -0,0 +1,60 @@
+#' Create a Survival Object from CDISC Data
+#'
+#' @description
+#' `r lifecycle::badge('experimental')`
+#'
+#' Create a survival object (e.g. `survival::Surv()`), which is usually used as
+#' the response variable in survival models.
+#' The function is designed to leverage the
+#' [CDISC ADaM ADTTE data model](https://www.cdisc.org/standards/foundational/adam/adam-basic-data-structure-bds-time-event-tte-analyses-v1-0)
+#' and more explicitly the conventions and controlled vocabulary of the data model.
+#' The `CNSR` argument expects a numeric column/vector that indicates
+#' an observation is censored.
+#'
+#' The `AVAL` and `CNSR` arguments are passed to
+#' `survival::Surv(time = AVAL, event = 1 - CNSR, type = 'right', origin = 0)`.
+#'
+#' @param AVAL The follow-up time. The follow-up time is assumed to originate from zero.
+#' When no argument is passed, the default value is a column/vector named `AVAL`
+#' @param CNSR The censoring indicator where `1=censored` and `0=death/event`.
+#' When no argument is passed, the default value is a column/vector named `CNSR`
+#'
+#' @return Object of class 'Surv'
+#' @seealso [`survival::Surv()`], [`estimate_KM()`]
+#' @export
+#'
+#' @examples
```

Comment:
can we have a more complex exampl? eg not KM related.

Reply:
Do you have something in mind?

I don't know if you were on the call during this part of the discussion. There is a bug in the current release of `coxph()` (that is now fixed, but not released) that does not allow for the use of `Surv_CDISC()` with `coxph()`.

Reply:
During the discussion the main argument was that this would be an amazing tool to do more advanced survival analysis (not KM, not cox). It would be good to show the user the advantage of having this function.

Reply:
I think we could show examples of the risks as motivating examples and then demonstrate workarounds currently being used, using say `broom::tidy`, `survminer`, any survival model in survival (i guess we need to wait for Coxph).  I'll forward some examples I've collected we could discuss to select from if we agree.

Reply:
I added an example of a parametric survival model and summarized the results with broom::tidy. I don't think we should show an example with survminor as that would add a package dependency.